pkg/discovery/oci_dbbacked.go

```diff
@@ -133,8 +139,48 @@ func (od *DBBackedOCIDiscovery) listGroupsFromInventory() ([]*plugininventory.Pl
 // inventory of this discovery and stores it in the cache directory.
 func (od *DBBackedOCIDiscovery) fetchInventoryImage() error {
 	// TODO(khouzam): Improve by checking if we really need to download again or if we can use the cache
+
+	// Get the custom public key path and prepare cosign verifier, if empty, cosign verifier would use embedded public key for verification
+	customPublicKeyPath := os.Getenv(constants.PublicKeyPathForPluginDiscoveryImageSignature)
+	cosignVerifier := cosignhelper.NewCosignVerifier(customPublicKeyPath)
+	if err := od.verifyInventoryImageSignature(cosignVerifier); err != nil {
+		log.Warningf("Unable to verify the plugins discovery image signature: %v", err)
+		// TODO(pkalle): Update the message to convey user to check if they could use the latest public key after we get details of the well known location of the public key
+		errMsg := fmt.Sprintf("Fatal, plugins discovery image signature verification failed. The `tanzu` CLI can not ensure the integrity of the plugins to be installed. To ignore this validation please set the environment variable %q to %q .  This is NOT RECOMMENDED and could put your environment at risk!",
```

Comment:
I like that you guide the user by telling them which value to set the env var to.
However, if we use the `TANZU_CLI_ADDITIONAL_PLUGIN_DISCOVERY_IMAGES_TEST_ONLY` variable, we could have more than one unsigned repos and then the user will need to add more than one to the SKIP variable, so this error message will mis-guide them.  How about:

```suggestion
		errMsg := fmt.Sprintf("Fatal, plugins discovery image signature verification failed. The `tanzu` CLI can not ensure the integrity of the plugins to be installed. To ignore this validation please append %q to the comma-separated list in the environment variable %q.  This is NOT RECOMMENDED and could put your environment at risk!",
```

Reply:
updated. Thanks